src/Microsoft.AspNetCore.Razor/Text/LineTrackingStringBuffer.cs

```diff
@@ -33,17 +33,24 @@ public SourceLocation EndLocation
 
         public void Append(string content)
         {
-            for (int i = 0; i < content.Length; i++)
+            var previousIndex = 0;
+            for (var i = 0; i < content.Length; i++)
             {
-                AppendCore(content[i]);
```

Comment:
Fewer `StringBuilder.ExpandByABlock` when we Append string segments rather than individual chars.